### PR TITLE
Hosting a sample ASP .NET Core in container using docker compose

### PR DESCRIPTION
**Pull request by ravipal**
Creating a new page on hosting a sample ASP .NET Core in container using docker compose

`@Rick-Anderson  edit:`
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/docker-compose-https?view=aspnetcore-2.1&branch=pr-en-us-17474)

### CODE REVIEW

File: aspnetcore/security/docker-compose-https.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+author: ravipal
+description: Learn how to host ASP.NET Core Images with Docker Compose over HTTPS
+monikerRange: '>= aspnetcore-2.1'
+ms.author: ravipal
+ms.custom: mvc
+ms.date: 07/05/2019
+no-loc: ["Let's Encrypt"]
+uid: security/docker-compose-https
+---
+# Hosting ASP.NET Core images with Docker over HTTPS
```
Comment (ghogen) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
# Hosting ASP.NET Core images with Docker Compose over HTTPS
```

File: aspnetcore/security/docker-compose-https.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
+---
+title: Hosting ASP.NET Core image in container using docker compose with HTTPS
+author: ravipal
+description: Learn how to host ASP.NET Core Images with Docker Compose over HTTPS
+monikerRange: '>= aspnetcore-2.1'
+ms.author: ravipal
+ms.custom: mvc
+ms.date: 07/05/2019
```
Comment (ravipal) [inline aspnetcore/security/docker-compose-https.md]: This date may need change. Should this be a current date?

File: aspnetcore/security/docker-compose-https.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+## Prerequisites
+
+The [.NET Core 2.2 SDK](https://dotnet.microsoft.com/download) or later is required for some of the instructions in this document.
+
+## Certificates
+
+A certificate from a [certificate authority](https://wikipedia.org/wiki/Certificate_authority) is required for [production hosting](https://blogs.msdn.microsoft.com/webdev/2017/11/29/configuring-https-in-asp-net-core-across-different-platforms/) for a domain. [Let's Encrypt](https://letsencrypt.org/) is a certificate authority that offers free certificates.
+
+This document uses [self-signed development certificates](https://en.wikipedia.org/wiki/Self-signed_certificate) for hosting pre-built images over `localhost`. The instructions are similar to using production certificates.
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
This document uses [self-signed development certificates](https://wikipedia.org/wiki/Self-signed_certificate) for hosting pre-built images over `localhost`. The instructions are similar to using production certificates.
```

File: aspnetcore/security/docker-compose-https.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+For production certs:
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
For production certificates:
```

File: aspnetcore/security/docker-compose-https.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+* The `dotnet dev-certs` tool is not required.
+* Certificates do not need to be stored in the location used in the instructions. Any location should work, although storing certs within your site directory is not recommended.
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
* Certificates don't need to be stored in the location used in the instructions. Any location should work, although storing certificates within your site directory is not recommended.
```

Comment (Rick-Anderson) [reply aspnetcore/security/docker-compose-https.md]: > Any location should work, although storing certificates within your site directory is not recommended.

This is saying two things. How about something like

Store the certificates in any location outside the site directory.

File: aspnetcore/security/docker-compose-https.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+The instructions contained in the following section volume mount certificates into containers using volumes: property in docker-compose.yml. You could add certificates into container images with a `COPY` command in a *Dockerfile*, but it's not recommended. Copying certificates into an image isn't recommended for the following reasons:
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
The instructions contained in the following section volume mount certificates into containers using the `volumes` property in *docker-compose.yml.* You could add certificates into container images with a `COPY` command in a *Dockerfile*, but it's not recommended. Copying certificates into an image isn't recommended for the following reasons:
```

File: aspnetcore/security/docker-compose-https.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+* It makes difficult to use the same image for testing with developer certificates.
+* It makes difficult to use the same image for Hosting with production certificates.
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
* It makes it difficult to use the same image for testing with developer certificates.
* It makes it difficult to use the same image for Hosting with production certificates.
```

File: aspnetcore/security/docker-compose-https.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+Generate certificate and configure local machine:
+
+```dotnetcli
+dotnet dev-certs https -ep %USERPROFILE%\.aspnet\https\aspnetapp.pfx -p { password here }
+dotnet dev-certs https --trust
+```
+
+In the preceding commands, replace `{ password here }` with a password.
+
+Create a docker-compose.debug.yml file with below content
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
Create a _docker-compose.debug.yml_ file with the following content:
```

File: aspnetcore/security/docker-compose-https.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+### macOS or Linux
+
+Generate certificate and configure local machine:
+
+```dotnetcli
+dotnet dev-certs https -ep ${HOME}/.aspnet/https/aspnetapp.pfx -p { password here }
+dotnet dev-certs https --trust
+```
+
+`dotnet dev-certs https --trust` is only supported on macOS and Windows. You need to trust certs on Linux in the way that is supported by your distro. It is likely that you need to trust the certificate in your browser.
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
`dotnet dev-certs https --trust` is only supported on macOS and Windows. You need to trust certificates on Linux in the way that is supported by your distro. It is likely that you need to trust the certificate in your browser.
```

File: aspnetcore/security/docker-compose-https.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+In the preceding commands, replace `{ password here }` with a password.
+
+Create a docker-compose.debug.yml file with below content
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
Create a _docker-compose.debug.yml_ file with the following content:
```

File: aspnetcore/security/docker-compose-https.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+Generate certificate and configure local machine:
+
+```dotnetcli
+dotnet dev-certs https -ep %USERPROFILE%\.aspnet\https\aspnetapp.pfx -p { password here }
+dotnet dev-certs https --trust
+```
+
+In the preceding commands, replace `{ password here }` with a password.
+
+Create a docker-compose.debug.yml file with below content
```
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
Create a _docker-compose.debug.yml_ file with the following content:
```

File: aspnetcore/security/docker-compose-https.md (hunk 2 above)
Comment (Rick-Anderson) [inline aspnetcore/security/docker-compose-https.md]:
```suggestion
ms.date: 03/28/2020
```

**Pull request merged**